bin/metrics/main.js

```diff
@@ -0,0 +1,124 @@
+const glob = require('glob');
+const jsc = require('jscodeshift');
+const flowParser = require('jscodeshift/parser/flow');
+const tsxParser = require('jscodeshift/parser/tsx');
+const {Lambda} = require('@aws-sdk/client-lambda');
+const prettier = require('prettier');
+const fs = require('fs');
+
+/* eslint-disable no-console */
+
+exports.main = async function (commitID, commitDate, {dry} = {}) {
+  const lambda = new Lambda({
+    region: 'eu-west-1',
+  });
+
+  if (!commitID) {
+    console.error('Invalid commit hash');
+    return;
+  }
+
+  if (!commitDate) {
+    console.error('Invalid date');
+    return;
+  }
+
+  const result = {
+    styleguideVersion: getVersion(),
+    commitID,
+    components: getComponents(),
+    commitDate: new Date(parseInt(commitDate, 10)).toISOString(),
+    project: getProject(),
+  };
+
+  if (dry) {
+    process.stdout.write(
+      `${prettier.format(JSON.stringify(result), {parser: 'json-stringify'})}`
+    );
+  } else {
+    await lambda
+      .invoke({
+        FunctionName: 'styleguide_metrics_post_components_lambda',
+        Payload: JSON.stringify(result),
+      })
+      .catch(e => {
+        console.error(e);
+      });
+  }
+};
+
+function getProject() {
+  const packageParsed = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
+
+  return packageParsed.name;
+}
+
+function getVersion() {
+  const packageParsed = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
+
+  return (
+    (packageParsed.dependencies &&
+      packageParsed.dependencies['brainly-style-guide']) ||
+    null
+  );
+}
+
+function getComponents() {
+  const elements = [];
+  const filePaths = glob.sync('**/*.{js,ts}x', {
+    ignore: 'node_modules/**',
+    cwd: process.cwd(),
```

Comment:
does it treat cwd as a root folder where search starts? We sometimes my need to search specific folder only (`src` etc)

Reply:
Yes. Now it search all folders from cwd. I can add option for providing search root paths. What do you think?

Reply:
Probably I would add it, in codemod cli we have it (but it is not immediately visible since it's passed to internal command)

Reply:
it can be normal argument even so:
`npx sg-metrics --dry src/*`

Reply:
cli currently requires 2 arguments, so it would be better to keep it as parameter like `--roots /root1 /root2`

Reply:
I'm also thinking about automatically retrieving commit hash and date. This could open way to have path as single parameter

Reply:
ok, it might be named parameter if you prefer that way, i would use `--path` rather than --roots though